modules/contact/src/actions/ContactAction.C

```diff
@@ -273,6 +273,22 @@ ContactAction::validParams()
       "Whether to use the Petrov-Galerkin approach for the mortar-based constraints. If set to "
       "true, we use the standard basis as the test function and dual basis as "
       "the shape function for the interpolation of the Lagrange multiplier variable.");
+
+  params.addParam<MooseEnum>(
+      "quadrature",
+      MooseEnum("DEFAULT FIRST SECOND THIRD FOURTH", "DEFAULT"),
+      "Quadrature rule to use on mortar segments. For 2D mortar DEFAULT is recommended."
+      "For 3D mortar, QUAD meshes are integrated using triangle mortar segments. "
+      "While DEFAULT quadrature order is typically sufficiently accurate, "
+      "exact integration of QUAD mortar faces requires SECOND order quadrature for FIRST variables "
+      "and FOURTH order quadrature for SECOND order variables.");
+
+  params.addParam<bool>("add_aux_vars",
```

Comment:
I would rename to generate_output and let the developers handle the minimum / maximum levels of outputs each formulation can handle

Reply:
generate_output is also the name used in the main SM action